### ISSUE TITLE
Convert AbstractPolyhedron to Polyhedra.polyhedron

### ISSUE THREAD

**Issue by schillic** [JuliaReach/LazySets.jl#1505]
Currently we can convert `HPolyhedron`, `HPolytope`, and `VPolytope` to `Polyhedra.polyhedron`. Since we can convert all `AbstractPolyhedron` types to `HPolyhedron` (and any `AbstractPolytope` type to all three of them), we can in principle also convert those types, but there is no elegant way.

#### Proposal
One of two options:
1. Make `polyhedron` a required interface method for `AbstractPolyhedron` types.
2. Since there are only two possible implementations (calling `vertices_list` or `constraints_list`), the only information we need is which of those two implementations is preferred. For example, an `AbstractHyperrectangle` would prefer `constraints_list`. This could be indicated by yet another helper method `is_hrep_preferred`, which might be helpful in other situations as well. Then the implementation is:
```julia
function polyhedron(P::AbstractPolyhedron)
    if is_hrep_preferred(P)
        return polyhedron(HPolyhedron(constraints_list(P)))
    else
        return polyhedron(VPolytope(vertices_list(P)))
    end
end
function is_hrep_preferred(P::AbstractPolyhedron)
    # default implementation
    return true
end

function is_hrep_preferred(P::VPolygon)
    return false
end
...
```

This is one open task in #1078.

**Comment by mforets** [comment]
FWIW the following two work: `polyhedron(convert(HPolytope, P))` or `polyhedron(convert(VPolytope, P))`, and i think this is simple enough to qualify as elegant :)

```julia
julia> using LazySets, Polyhedra
[ Info: Recompiling stale cache file C:\Users\marce\.julia\compiled\v1.2\Polyhedra\17i4E.ji for Polyhedra [67491407-f73d-577b-9b50-8179a7c68029]

julia> P = rand(Hyperrectangle)
Hyperrectangle{Float64,Array{Float64,1},Array{Float64,1}}([0.9714024861824725, -1.5203513209265647], [1.0954915952635476, 0.7449180042707249])

julia> convert(HPolytope, P) |> polyhedron
Polyhedron DefaultPolyhedron{Float64,MixedMatHRep{Float64,Array{Float64,2}},MixedMatVRep{Float64,Array{Float64,2}}}:
4-element iterator of Polyhedra.HalfSpace{Float64,Array{Float64,1}}:
 HalfSpace([1.0, 0.0], 2.06689408144602)
 HalfSpace([0.0, 1.0], -0.7754333166558398)
 HalfSpace([-1.0, 0.0], 0.12408910908107507)
 HalfSpace([0.0, -1.0], 2.2652693251972895)

julia> convert(VPolytope, P) |> polyhedron
Polyhedron DefaultPolyhedron{Float64,Polyhedra.Intersection{Float64,Array{Float64,1},Int64},Polyhedra.Hull{Float64,Array{Float64,1},Int64}}:
4-element iterator of Array{Float64,1}:
 [2.06689408144602, -0.7754333166558398]
 [-0.12408910908107507, -0.7754333166558398]
 [2.06689408144602, -2.2652693251972895]
 [-0.12408910908107507, -2.2652693251972895]
```

**Comment by mforets** [comment]
In `function is_hrep_preferred(P::AbstractPolyhedron)` i would use as argument to the *type* of `P` instead of the value.

**Comment by schillic** [comment]
> FWIW the following two work: polyhedron(convert(HPolytope, P)) or polyhedron(convert(VPolytope, P)), and i think this is simple enough to qualify as elegant :)

The argument for having the proposed function is twofold:
1. As a user you have to know the most efficient way.
2. In a library function `f` you would have to dispatch on the input, which means for efficient code you need to add 20 methods of `f` just to choose the most efficient conversion. The dispatch should happen on simple helper functions rather than on the high-level code.

**Comment by schillic** [comment]
This was resolved in #3223.
